general/mem_manager.cpp

```diff
@@ -869,37 +869,40 @@ void MemoryManager::SetDeviceMemoryType_(void *h_ptr, unsigned flags,
    }
 }
 
-MemoryType MemoryManager::Delete_(void *h_ptr, MemoryType mt, unsigned flags)
+MemoryType MemoryManager::Delete_(void *h_ptr, MemoryType h_mt, unsigned flags)
```

Comment:
Was the assumption that mt had to be a h_mt?

Reply:
We only call it with `Memory::h_mt`, so I just made this more explicit.